source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/CustomProgressBarRenderer.mm

```diff
@@ -0,0 +1,35 @@
+//
+//  CustomProgressBarRenderer
+//  CustomProgressBarRenderer.mm
+//
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import "CustomProgressBarRenderer.h"
+
+@implementation CustomProgressBarRenderer
+
+- (UIView *)DeserializeToCustomUIElement:(NSData* )json
+{
+        if(json) {
+            NSArray *data = [NSJSONSerialization JSONObjectWithData:json options:NSJSONReadingMutableLeaves error:nil];
+            NSDictionary<NSString *, NSNumber *> *myval = [data objectAtIndex:0];
+            UIProgressView *progressView;
+            progressView = [[UIProgressView alloc] initWithProgressViewStyle:UIProgressViewStyleBar];
+            NSNumber *red = myval[@"red"];
```

Comment:
Looks good!

Reply:
thanks

---
In reply to: [172225868](https://github.com/Microsoft/AdaptiveCards/pull/1211#discussion_r172225868) [](ancestors = 172225868)